polaris/dataset/_dataset.py

```diff
@@ -516,8 +516,8 @@ def __getitem__(self, item):
 
     def _repr_dict_(self) -> dict:
         """Utility function for pretty-printing to the command line and jupyter notebooks"""
-        repr_dict = self.model_dump()
-        repr_dict.pop("table")
+        repr_dict = self.model_dump(exclude={"table"})
+        repr_dict["cache_dir"] = str(repr_dict["cache_dir"])
```

Comment:
**Suggestion:** Add a [`@field_serializer`](https://docs.pydantic.dev/latest/api/functional_serializers/#pydantic.functional_serializers.field_serializer) for the `cache_dir` to the `Dataset` class. That way, we centralize the logic for serialization.